upup/pkg/kutil/rollingupdate_cluster.go

```diff
@@ -269,9 +270,42 @@ func buildCloudInstanceGroup(ig *api.InstanceGroup, g *autoscaling.Group, nodeMa
 	return n
 }
 
-func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, k8sClient *k8s_clientset.Clientset) error {
+func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval time.Duration, headroom int64, k8sClient *k8s_clientset.Clientset) error {
 	c := cloud.(awsup.AWSCloud)
 
+	if headroom > 0 {
+		describeParams := &autoscaling.DescribeAutoScalingGroupsInput{
+			AutoScalingGroupNames: []*string{aws.String(n.ASGName)},
+			MaxRecords: aws.Int64(1),
+		}
+
+		out, err := c.Autoscaling().DescribeAutoScalingGroups(describeParams)
+		if err != nil {
+			return fmt.Errorf("Error adding headroom to ASG %s: %v", n.ASGName, err)
+		}
+
+		currentCapacity := aws.Int64Value(out.AutoScalingGroups[0].DesiredCapacity)
+		desiredCapacity := currentCapacity + headroom
+
+		if desiredCapacity > *out.AutoScalingGroups[0].MaxSize {
+			return fmt.Errorf("Adding headroom exceeds ASG max, increase ASG max to continue.")
```

Comment:
So most people today are not (yet) running the cluster autoscaler - I would love to work with you on that though @andrewsykim !

So most people actually have min == max == desired = N, and wouldn't be able to use headroom.

I also wonder how this interacts with the cluster-autoscaler.  Presumably the autoscaler would terminate some pods soon after we added the "headroom" pods.  

A few options I think:

* We could move to a 2 ASG approach (high impact)
* We could integrate with cluster autoscaler and "hand off" some work to it?   (waving hands, and this is probably hard)
* We could raise the max in this circumstance, and do our best to set it back before exiting.  A rolling update could also resync the min & max sizes, in case a previous rolling-update failed.

Not entirely sure how best to handle this, but I think it's a great feature I'd love to see in.  Any thoughts on which is the best option here?